routes/api/tag-routes.js

```diff
@@ -1,28 +1,57 @@
-const router = require('express').Router();
-const { Tag, Product, ProductTag } = require('../../models');
+const router = require("express").Router();
+const { Tag, Product, ProductTag } = require("../../models");
 
 // The `/api/tags` endpoint
 
-router.get('/', (req, res) => {
+router.get("/", async (req, res) => {
+  const all = await Tag.findAll();
+  res.send(all);
   // find all tags
   // be sure to include its associated Product data
 });
 
-router.get('/:id', (req, res) => {
+router.get("/:id", async (req, res) => {
   // find a single tag by its `id`
+  const tag = await Tag.findByPk(req.params.id);
+  res.send(tag);
   // be sure to include its associated Product data
 });
 
-router.post('/', (req, res) => {
+router.post("/", async (req, res) => {
   // create a new tag
+  const newTag = await Tag.create({
+    tag_name: req.body.tag_name,
+  });
+  res.send(newTag);
 });
 
-router.put('/:id', (req, res) => {
+router.put("/:id", async (req, res) => {
   // update a tag's name by its `id` value
+  const updateTag = await Tag.update(
+    {
+      tag_name: req.body.tag_name,
```

Comment:
consider adding some validation wherever you;re reading data from the body so that an operation is only performed if the user has provided all of the required data within the body of the request. This will help to further protect your app from breakages :-)